### ISSUE TITLE
When I call PollNextOverlayEvent() function, VREvent_MouseMove, VREvent_MouseButtonDown, VREvent_MouseButtonUp events don't work.

### ISSUE THREAD

**Issue by SGHTechdiv** [ValveSoftware/openvr#712]
So, I've been working on application that has a overlay pop-up feature. Before the SteamVR update, it worked all fine. But since the last update of SteamVR which was yesterday(Jan/10/2018 KST), when I point the controller on overlay pop-up, PollNextOverlayEvent() functions is called but VREvent_MouseMove, VREvent_MouseButtonDown, VREvent_MouseButtonUp events are not occurred. 
It seems like an abnormal situation and I'd like to request you guys to check out this issue. 

(FYI, I applied Unity SDK(v1.2.3) as well on my application.)

**Comment by bddckr** [comment]
This may be an oversight by the latest release. All I can find is this:
> IVROverlay
> * Removed HandleControllerOverlayInteractionAsMouse. Overlay applications should use the mouse overlay input method instead.

You may try to set the input method specifically using `IVROverlay.SetOverlayInputMethod(ulOverlayHandle, VROverlayInputMethod.VROverlayInputMethod_Mouse)`.

**Comment by SGHTechdiv** [comment]
The function you mentioned, IVROverlay.SetOverlayInputMethod(ulOverlayHandle, VROverlayInputMethod.VROverlayInputMethod_Mouse), is being called. 
Yet, the events don't occur.

**Comment by JoeLudwig** [comment]
Is this a dashboard overlay or a normal overlay? You're setting the input type for the overlay to Mouse?

Do you expect the full laser mouse input mode in this case (i.e. controllers are drawn, the laser pointer and blue blob are drawn, and the game itself doesn't get input)?  Or do you just want events while still processing other input in the app itself?

**Comment by SGHTechdiv** [comment]
(Using Unity 5.6.0)
with OpenVR.Overlay.CreateOverlay() , I created a normal overlay pop-up.
And let it drawn by using OpenVR.Overlay.ShowOverlay() function.

It is similar to <class SteamVR_Overlay> code in Unity SteamVR SDK (v1.2.3).
<VROverlayInputMethod.Mouse> settings are already configured.
I configured most settings referring to <class SteamVR_Overlay> code.

Controller input normally works. (All inputs by controller are normal.)
And I know that the laser point of controller doesn't cause on normal overlay.

I've been implementing some UIs on normal overlay.
What I need is, the location information of where controller is pointing.
just like a mouse cursor feature in PC website.

Before SteamVR update, when I point something or pull the trigger with controller,
<EVREventType.VREvent_MouseMove, VREvent_MouseButtonDown, VREvent_MouseButtonUp> Events occur thanks to function <OpenVR.Overlay.PollNextOverlayEvent()>
This enable me to make UI cursor and button click feature.

After the SteamVR update, those events don't occur.
Please go over the issue and let me kindly ask you to fix it. 

If anything is unclear, let me know.

**Comment by GimpMaster** [comment]
@JoeLudwig I have this same issue. I have a standard overlay (not dashboard) and I want the laser pointer and blue blob to work. Before January 5th update it worked great. Now it no longer draws laser or blob and No events.

**Comment by SGHTechdiv** [comment]
Anything working on with this issue???

**Comment by GimpMaster** [comment]
@SGHTechdiv no. It's still broken. What is more disturbing is the root of the issue seems to be the following....

1. Dont run any of your own software
2. Open Steam VR settings and go to Developer section
3. Disable Steam Dashboard
4. Click the System Menu button.
5. Watch your controllers go completely non-functional and your screen gets blurred.

It's almost like it's still trying to bring up the steam dashboard. This is not fixed in any of the new betas.

@JoeLudwig @johnv-valve  @alexvalve @jeffmvalve

**Comment by GimpMaster** [comment]
@JoeLudwig @johnv-valve  @alexvalve  @jeffmvalve 

With the latest beta on Steam VR released tonight this still doesn't work. Overlays haven't been able to receive mouse input since December. Any thoughts?

**Comment by GimpMaster** [comment]
Let me clarify this a little bit more 

VRApplication_Overlay - works just fine. 

vr::VRApplication_Background - do not

**Comment by GimpMaster** [comment]
@hyungjunk - unfortunately it seems Valve does not plan on supporting background application overlay mouse interaction any time soon. @JoeLudwig  mentioned that it worked before on mistake and that it will take some time to put it back in. I've already figured out another means of input using the trackpad for selection.

**Comment by bddckr** [comment]
I'd love to get back laser beam and interaction events from the pointer even on non-dashboard overlays if possible. Maybe this functionality could be activated by a new flag, but past-me thought it would be enough to just set the input method to "mouse" because that is something we do proactively already. 😄

**Comment by tarung27** [comment]
Is this issue resolved yet?  @GimpMaster _"I've already figured out another means of input using the trackpad for selection."_ What is the workaround your tried.  Wondering if it is possible to shoot laser beam with trackpad?

**Comment by bddckr** [comment]
Nothing changed. It doesn't look like Valve are interested in bringing back this functionality.

You can probably get laser pointers to work yourself by utilizing stereo overlays and drawing the beam yourself into those textures. I never went down that route as it's a time sink.

The trackpad solution probably just means listening for the button presses and routing that to your UI giving it navigation and selection logic that way.

**Comment by tarung27** [comment]
@bddckr Thanks.  I have a scene application where I need small UI popups.  How about drawing Quads as part of the scene with textures as an alternative? Of course, will have to handle intersection on my own to simulate UI events.
